src/operation-node/join-node.ts

```diff
@@ -15,7 +16,7 @@ export interface JoinNode extends OperationNode {
   readonly kind: 'JoinNode'
   readonly joinType: JoinType
   readonly table: OperationNode
-  readonly on?: OnNode
+  readonly on?: OnNode | JoinUsingNode
```

Comment:
That's a breaking change not worth doing, and also a surprise - how's `on` related to `using`?

This should make things quite simple, and no need for an extra node type.

```suggestion
  readonly on?: OnNode
  readonly using?: ReadonlyArray<ColumnNode>
```